core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -160,47 +135,38 @@ private static String getUnixSocketArg(ConnectionConfig config) {
    * @return the newly created Socket.
    * @throws IOException if error occurs during socket creation.
    */
-  public static Socket connect(Properties props) throws IOException, InterruptedException {
+  public Socket connect(Properties props) throws IOException, InterruptedException {
```

Comment:
Since we own this API, why not just have the connect that takes a `ConnectionConfig`?

Reply:
I'll remove this and make the SocketFactory implementations pass a ConnectionConfig instead.